File: src/Administration/Reservation.java
package Administration;

import Animals.Animal;
import Animals.Cat;
import Animals.Dog;
import Animals.Gender;

import java.util.ArrayList;
import java.util.Observable;

public class Reservation extends Observable
{
    public ArrayList<Animal> Animals = new ArrayList<Animal>();

    public Reservation()
    {

    }

    public void NewCat(String name, Gender gender, String badHabits)
    {
        Animal newcat = new Cat(name, gender, badHabits);
        this.Animals.add(newcat);
        setChanged();
        notifyObservers(newcat);
    }

    public void NewDog(String name, Gender gender)
    {
        Animal newdog = new Dog(name, gender);
        this.Animals.add(new Dog(name, gender));
        setChanged();
        notifyObservers(newdog);
    }
}

File: Animal Shelter Opdracht/C# versie/animal_shelter-abstract_interfaces-649f03d658d7546000a2cadc0e118ada4f610607/Animal Shelter/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Animal_Shelter.Animals;

namespace Animal_Shelter
{
    public class Reservation
    {
        public List<Animal> Animals { get; private set; } = new List<Animal>();
        
        public void NewCat(string name, Gender gender, string badHabits)
        {
            this.Animals.Add(new Cat(name, gender, badHabits));
        }
        
        public void NewDog(string name, Gender gender)
        {
            this.Animals.Add(new Dog(name, gender));
        }
    }
}
